x/gamm/types/errors.go

```diff
@@ -16,18 +16,19 @@ func (e PoolDoesNotExistError) Error() string {
 
 // x/gamm module sentinel errors.
 var (
-	ErrPoolNotFound        = sdkerrors.Register(ModuleName, 1, "pool not found")
-	ErrPoolAlreadyExist    = sdkerrors.Register(ModuleName, 2, "pool already exist")
-	ErrPoolLocked          = sdkerrors.Register(ModuleName, 3, "pool is locked")
-	ErrTooFewPoolAssets    = sdkerrors.Register(ModuleName, 4, "pool should have at least 2 assets, as they must be swapping between at least two assets")
-	ErrTooManyPoolAssets   = sdkerrors.Register(ModuleName, 5, "pool has too many assets (currently capped at 8 assets per balancer pool and 2 per stableswap)")
-	ErrLimitMaxAmount      = sdkerrors.Register(ModuleName, 6, "calculated amount is larger than max amount")
-	ErrLimitMinAmount      = sdkerrors.Register(ModuleName, 7, "calculated amount is lesser than min amount")
-	ErrInvalidMathApprox   = sdkerrors.Register(ModuleName, 8, "invalid calculated result")
-	ErrAlreadyInvalidPool  = sdkerrors.Register(ModuleName, 9, "destruction on already invalid pool")
-	ErrInvalidPool         = sdkerrors.Register(ModuleName, 10, "attempting to create an invalid pool")
-	ErrDenomNotFoundInPool = sdkerrors.Register(ModuleName, 11, "denom does not exist in pool")
-	ErrDenomAlreadyInPool  = sdkerrors.Register(ModuleName, 12, "denom already exists in the pool")
+	ErrPoolNotFound         = sdkerrors.Register(ModuleName, 1, "pool not found")
+	ErrPoolAlreadyExist     = sdkerrors.Register(ModuleName, 2, "pool already exist")
+	ErrPoolLocked           = sdkerrors.Register(ModuleName, 3, "pool is locked")
+	ErrTooFewPoolAssets     = sdkerrors.Register(ModuleName, 4, "pool should have at least 2 assets, as they must be swapping between at least two assets")
+	ErrTooManyPoolAssets    = sdkerrors.Register(ModuleName, 5, "pool has too many assets (currently capped at 8 assets per balancer pool and 2 per stableswap)")
+	ErrLimitMaxAmount       = sdkerrors.Register(ModuleName, 6, "calculated amount is larger than max amount")
+	ErrLimitMinAmount       = sdkerrors.Register(ModuleName, 7, "calculated amount is lesser than min amount")
+	ErrInvalidMathApprox    = sdkerrors.Register(ModuleName, 8, "invalid calculated result")
+	ErrAlreadyInvalidPool   = sdkerrors.Register(ModuleName, 9, "destruction on already invalid pool")
+	ErrInvalidPool          = sdkerrors.Register(ModuleName, 10, "attempting to create an invalid pool")
+	ErrDenomNotFoundInPool  = sdkerrors.Register(ModuleName, 11, "denom does not exist in pool")
+	ErrDenomAlreadyInPool   = sdkerrors.Register(ModuleName, 12, "denom already exists in the pool")
+	ErrInvalidNumberOfCoins = sdkerrors.Register(ModuleName, 13, "balancer pool only supports LP'ing with one asset or all assets in pool")
```

Comment:
I suggest to consider defining a Go native error instead of SDK error. We are trying to move away from using SDK errors.

Reply:
The error is originally a Go native error, but I changed it to match the other tests in the table.  The other tests match SDK errors, and line 161, https://github.com/RusAkh/osmosis-fork/blob/issue2956/x/gamm/keeper/grpc_query_test.go#L160, requires the error to be registered.  The last test in the join pool test table would return the error: `Target error should be in err chain` when returning a Go error: https://github.com/RusAkh/osmosis-fork/blob/issue2956/x/gamm/keeper/grpc_query_test.go#L135.

Reply:
Sounds good, thanks for clarifying. We will refactor that separately when deprecating skderrors

Reply:
> LGTM. I would like one more reviewer to approve this before merging please

Awesome, thank you.